src/main/js/store/actions/distributionConfigs.js

```diff
@@ -206,11 +238,19 @@ export function testDistributionJob(config) {
         request.then((response) => {
             response.json()
                 .then((responseData) => {
-                    if (response.ok) {
+                    const defaultHandler = () => jobError({
+                        type: DISTRIBUTION_JOB_TEST_FAILURE,
+                        ...responseData
+                    });
+                    const handler = createErrorHandler(DISTRIBUTION_JOB_TEST_FAILURE, defaultHandler);
+                    if (responseData.errors && !Object.keys(responseData.errors).length) {
                         dispatch(testJobSuccess(responseData.message));
+                    } else if (!response.ok) {
+                        dispatch(handler(response.status));
                     } else {
-                        handleFailureResponse(DISTRIBUTION_JOB_TEST_FAILURE, dispatch, responseData, response.status);
+                        dispatch(handler(400));
                     }
+                    dispatch(handler(response.status));
```

Comment:
What's the difference among these three calls? It seems like we should be able to drop the else-if and else to just call `dispatch(handler(response.status));`

Reply:
the dispatch at the end I need to remove